src/javascript/Edit/engineTabs/useRegisterEngineTabActions.js

```diff
@@ -0,0 +1,57 @@
+import {useEffect} from 'react';
+import {Constants} from '~/ContentEditor.constants';
+import {engineTabsPermissionCheckQuery} from '~/Edit/engineTabs/engineTabs.permission.gql-query';
+import {registry} from '@jahia/ui-extender';
+import openEngineTabs from '~/Edit/engineTabs/openEngineTabs.action';
+import {getNodeTypes} from './engineTabs.utils';
+import {useQuery} from '@apollo/react-hooks';
+import {useContentEditorContext} from '~/ContentEditor.context';
+
+/**
+ * This function register the actions related to the GWT engine tabs
+ */
+export const useRegisterEngineTabActions = () => {
+    const {nodeData, site} = useContentEditorContext();
+    const {path, displayName, uuid, mixinTypes, primaryNodeType} = nodeData;
+
+    // Get tabs from GWT authoring API
+    let tabs = window.authoringApi.getEditTabs(
```

Comment:
```suggestion
    const tabs = window.authoringApi.getEditTabs(
```

I'm happy to see custom hook :)

Reply:
Don't forget to unit test this function.